src/TLuaInterpreter.cpp

```diff
@@ -2742,61 +2677,54 @@ int TLuaInterpreter::disableCommandLine(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#replace
 int TLuaInterpreter::replace(lua_State* L)
 {
-    std::string a1 = "";
-    std::string a2 = "";
     int n = lua_gettop(L);
     int s = 1;
     if (!lua_isstring(L, s)) {
         lua_pushstring(L, "replace: wrong argument type");
         lua_error(L);
         return 1;
-    } else {
-        a1 = lua_tostring(L, s);
-        s++;
     }
+    QString name = lua_tostring(L, s);
+    s++;
 
-    QString _name(a1.c_str());
-    std::string luaSendText = "";
+    QString text = "";
     if (n > 1) {
         if (!lua_isstring(L, s)) {
             lua_pushstring(L, "replace: wrong argument type");
             lua_error(L);
             return 1;
-        } else {
-            a2 = lua_tostring(L, s);
         }
+        text = lua_tostring(L, s);
     }
 
     Host& host = getHostFromLua(L);
     if (n == 1) {
-        host.mpConsole->replace(QString(a1.c_str()));
+        host.mpConsole->replace(text);
```

Comment:
Whoops, this should be `name` instead..!

Your named variables are a bit confusing, because argument #1 can be optional.
If there is only one argument, then it will be text, otherwise the window name.
Maybe use a name like `name_or_text` for the first argument instead?

Reply:
Thans. Will fix.